walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,63 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
 
 public final class FindMeetingQuery {
+  /**
+   * Returns the times when {@code request} could happen on a day containing {@code events}
+   * Time complexity of the algorithm is O(nlogn + mn), in which
+   * n is the number of {@code events}, and m is the number of attendees in {@code request}
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    Collection<TimeRange> availableTimes = new ArrayList<>();
+    List<Event> eventList = new ArrayList<>(events);
+    eventList.sort(Comparator.comparingInt(e -> e.getWhen().start()));
```

Comment:
Just FYI, there's also a provided `Comparator` for this.
`com.google.sps.TimeRange#ORDER_BY_START`

Reply:
The Comparator is for TimeRange, thus i implemented a new one for Event.

Reply:
Ah noted. I missed the type that you're comparing.